realm/realm-library/src/main/java/io/realm/AndroidNotifier.java

```diff
@@ -0,0 +1,166 @@
+import io.realm.internal.log.RealmLog;
+
+/**
+ * Implementation of {@link RealmNotifier} for Android based on {@link Handler}.
+ */
+// FIXME: Please move me to the io.realm.internal when HandlerController is there.
+class AndroidNotifier implements RealmNotifier {
+    private Handler handler;
+
+    public AndroidNotifier(HandlerController handlerController) {
+        if (isAutoRefreshAvailable()) {
+            handler = new Handler(handlerController);
+        }
+    }
+
+    // Called by Java when transaction committed to send LOCAL_COMMIT to current thread's handler.
+    @Override
+    public void notifyByLocalThread() {
+        if (handler == null) return;
+
+        // Force any updates on the current thread to the front the queue. Doing this is mostly
+        // relevant on the UI thread where it could otherwise process a motion event before the
+        // REALM_CHANGED event. This could in turn cause a UI component like ListView to crash. See
+        // https://github.com/realm/realm-android-adapters/issues/11 for such a case.
+        // Other Looper threads could process similar events. For that reason all looper threads will
+        // prioritize local commits.
+        //
+        // If a user is doing commits inside a RealmChangeListener this can cause the Looper thread to get
+        // event starved as it only starts handling Realm events instead. This is an acceptable risk as
+        // that behaviour indicate a user bug. Previously this would be hidden as the UI would still
+        // be responsive.
+        Message msg = Message.obtain();
+        msg.what = HandlerControllerConstants.LOCAL_COMMIT;
+        if (!handler.hasMessages(HandlerControllerConstants.LOCAL_COMMIT)) {
+            handler.removeMessages(HandlerControllerConstants.REALM_CHANGED);
+            handler.sendMessageAtFrontOfQueue(msg);
+        }
+    }
+
+    // This is called by OS when other thread/process changes the Realm.
+    // This is getting called on the same thread which creates the Realm.
+    // FIXME: The whole calling routine is twisted and needs to be rewritten in the near future.
+    // |---------------------------------------------------------------+--------------+------------------------------------------------|
+    // | Thread A                                                      | Thread B     | Daemon Thread                                  |
+    // |---------------------------------------------------------------+--------------+------------------------------------------------|
+    // |                                                               | Make changes |                                                |
+    // |                                                               |              | Detect and notify thread A through JNI ALooper |
+    // | Call Realm::notify() in ALooper callback                      |              |                                                |
```

Comment:
Perhaps clarify that this is happening in the OS? I was looking for the `notify()` method on the Java Realm.